src/libengine/engine_bind.cc

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "src/libengine/engine_bind.h"
+
+#include <openssl/engine.h>
+
+#include "src/libengine/engine_name.h"
+#include "src/libengine/engine_setup.h"
+
+namespace libengine {
```

Comment:
I'd leave off the `lib`: `engine` instead of `libengine`

Reply:
I added a `kmsengine` top-level namespace and renamed `namespace libengine` to `namespace engine`, though not sure if there'll be confusion between `engine` and `kmsengine`.

Reply:
If it's more mnemonic as to how the code flows, we could call the two .so namespaces `frontend` and `backend`, or something else that sacrifices the "engine" naming for a more directional flavor. Not sure if it really matters though, but probably easier to rename things now before more files get added.

Reply:
What's the difference between `kmsengine` and `engine`? 😅

`::kmsengine::engine::DoFoo()` seems a bit awkward?

Reply:
See discussion in this thread above - we discussed in the design-doc adding a top-level namespace shared between both libengine and libkms, and then in this thread we decided to rename `namespace libengine` to `namespace engine`. I wasn't sure if there would be confusion between `engine` and `kmsengine`, so maybe another name to replace `engine` is in order.

Reply:
You could rename `engine` `openssl_interface` or something.